src/GLdispatch/vnd-glapi/entry_ppc64le_tsd.c

```diff
@@ -35,24 +35,68 @@
        ".hidden public_entry_start\n"
         "public_entry_start:");
 
-#define STUB_ASM_ENTRY(func)        \
-   ".globl " func "\n"              \
-   ".type " func ", @function\n"    \
-   ".balign " U_STRINGIFY(PPC64LE_ENTRY_SIZE) "\n"                   \
-   func ":"
-
-#define STUB_ASM_CODE(slot) \
-    "nop"
-    // TODO: Fill in this assembly code.
+#define STUB_ASM_ENTRY(func)                            \
+    ".globl " func "\n"                                 \
+    ".type " func ", @function\n"                       \
+    ".balign " U_STRINGIFY(PPC64LE_ENTRY_SIZE) "\n"     \
+    func ":\n\t"                                        \
+    "  addis  2, 12, .TOC.-" func "@ha\n\t"             \
+    "  addi   2, 2, .TOC.-" func "@l\n\t"               \
+    "  .localentry  " func ", .-" func "\n\t"
+
+#define STUB_ASM_CODE(slot)                             \
+    "  addis  11, 2, _glapi_Current@got@ha\n\t"         \
+    "  ld     11, _glapi_Current@got@l(11)\n\t"         \
+    "  ld     11, 0(11)\n\t"                            \
+    "  cmpldi 11, 0\n\t"                                \
+    "  beq    2000f\n"                                  \
+    "1050:\n\t"                                         \
+    "  ld     12, " slot "*8(11)\n\t"                   \
+    "  mtctr  12\n\t"                                   \
+    "  bctr\n"                                          \
+    "2000:\n\t"                                         \
+    "  mflr   0\n\t"                                    \
+    "  std    0, 16(1)\n\t"                             \
+    "  std    2, 40(1)\n\t"                             \
+    "  stdu   1, -144(1)\n\t"                           \
+    "  std    3, 56(1)\n\t"                             \
+    "  stq    4, 64(1)\n\t"                             \
+    "  stq    6, 80(1)\n\t"                             \
+    "  stq    8, 96(1)\n\t"                             \
+    "  std    10, 112(1)\n\t"                           \
+    "  std    12, 128(1)\n\t"                           \
+    "  addis  12, 2, _glapi_get_current@got@ha\n\t"     \
+    "  ld     12, _glapi_get_current@got@l(12)\n\t"     \
+    "  mtctr  12\n\t"                                   \
+    "  bctrl\n\t"                                       \
+    "  ld     2, 144+40(1)\n\t"                         \
+    "  mr     11, 3\n\t"                                \
+    "  ld     3, 56(1)\n\t"                             \
+    "  lq     4, 64(1)\n\t"                             \
+    "  lq     6, 80(1)\n\t"                             \
+    "  lq     8, 96(1)\n\t"                             \
+    "  ld     10, 112(1)\n\t"                           \
+    "  ld     12, 128(1)\n\t"                           \
+    "  addi   1, 1, 144\n\t"                            \
+    "  ld     0, 16(1)\n\t"                             \
+    "  mtlr   0\n\t"                                    \
+    "  b      1050b\n"
```

Comment:
As usual, I don't feel qualified to review these but I'll take your word for it that they work.